src/Microsoft.OData.Edm/EdmUtil.cs

```diff
@@ -272,17 +275,25 @@ internal static bool TryGetNamespaceNameFromQualifiedName(string qualifiedName,
                 {
                     namespaceName = String.Empty;
                     name = qualifiedName;
-                    return false;
-                }
 
-                namespaceName = qualifiedName.Substring(0, lastDot);
-                name = qualifiedName.Substring(lastDot + 1);
-                return true;
+                    foundNamespace = false;
+                }
+                else
+                {
+                    namespaceName = qualifiedName.Substring(0, lastDot);
+                    name = qualifiedName.Substring(lastDot + 1);
+                    foundNamespace = true;
+                }
+            }
+            else
+            {
+                foundNamespace = true;
+                namespaceName = qualifiedName.Substring(0, lastSlash);
+                name = qualifiedName.Substring(lastSlash + 1);
             }
 
-            namespaceName = qualifiedName.Substring(0, lastSlash);
-            name = qualifiedName.Substring(lastSlash + 1);
-            return true;
+            fullName = EdmUtil.GetFullNameForSchemaElement(namespaceName, name);

```

Comment:
From looking at each of the if-else statements above, I think that the `GetFullNameForSchemaElement` function is effectively changing any '/' in the qualifiedName to '.'. In all other cases, `fullName` seems to be equal to `qualifiedName`.

Please correct me if I'm wrong, but is `qualifiedName` essentially the "full type name" that you're looking for? If so, you can cache the `qualifiedName` instead of calculating the `fullName`.

Reply:
I don't believe this is completely accurate. FullName is supposed to be delimited by "." instead of "/".

This is to keep backward compatibility. I am afraid I don't know why qualified name is with "/" and full Name is with "." 

I am attempting a perf fix so afraid to do functional changes.